Dockerfile

```diff
@@ -14,14 +14,14 @@ COPY . .
 ENV CGO_ENABLED=0
 
 # Note the version MUST MATCH otelarrowcol-build.yaml
-RUN go install go.opentelemetry.io/collector/cmd/builder@v0.89.0
+RUN go install go.opentelemetry.io/collector/cmd/builder@v0.140.0
```

Comment:
Ah! This is correct in the sense that the collector/otelarrowcol-build.yaml already refers to a v0.140.x series of components. However, it means a previous change modified the component versions but not the cmd/builder version, and these genuinely need to match (i.e., the "MUST MATCH" comment above was not followed!).

There is also a `make` rule that I suppose you could invoke (i.e., `make otelarrowcol`) which checks that the code base is up-to-date with respect to the commands on line 22-24. If the source code is up-to-date we shouldn't need these steps.